### PR TITLE
fix(staking): page inconsistencies

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.monday.com/boards/2218344956/pulses/2273556093

### CODE REVIEW

**Comment by ShynRou** [comment]
Looks pretty good, but there is still a state where the vests table is displayed as empty, it should either show a loading text or the you have not vests text.
image

**Review by tiltom — APPROVED** [review]

**Review by ShynRou — CHANGES_REQUESTED** [review]
See comment above

**Review by JoeStorm99 — APPROVED** [review]

File: src/app/containers/StakePage/components/VestingContract.tsx [hunk 1]
```diff
@@ -168,145 +168,134 @@ export function VestingContract(props: Props) {
...
+            <div>
+              <img src={logoSvg} className="tw-ml-3 tw-mr-3" alt="sov" />
+            </div>
+            <div className="tw-text-sm tw-font-normal tw-hidden xl:tw-block tw-pl-3">
+              {t(translations.stake.currentVests.assetType[props.type])}
+            </div>
+          </div>
+        </td>
+        <td className="tw-text-left tw-font-normal">
+          <p className={`tw-m-0 ${lockedAmount.loading && 'tw-skeleton'}`}>
```
Comment (tiltom) [inline src/app/containers/StakePage/components/VestingContract.tsx:183]:
```suggestion
          <p className={classNames('tw-m-0', {
          'tw-skeleton': lockedAmount.loading }
          )}>
```

File: src/app/containers/StakePage/components/VestingContract.tsx [hunk 2]
```diff
@@ -168,145 +168,134 @@ export function VestingContract(props: Props) {
...
+                <LoadableValue
+                  value={weiToUSD(dollarValue)}
+                  loading={dollars.loading}
+                />
+              </>
+            )}
+          </p>
+        </td>
+        <td className="tw-text-left tw-hidden lg:tw-table-cell tw-font-normal">
+          <p className={`tw-m-0 ${delegateLoading && 'tw-skeleton'}`}>
```
Comment (tiltom) [inline src/app/containers/StakePage/components/VestingContract.tsx:197]:
```suggestion
          <p className={classNames('tw-m-0', {
           'tw-skeleton': delegateLoading}
           )}>
```

File: src/app/containers/StakePage/components/VestingContract.tsx [hunk 3]
```diff
@@ -168,145 +168,134 @@ export function VestingContract(props: Props) {
...
+            {delegate.length > 0 && (
+              <>
+                <AddressBadge
+                  txHash={delegate}
+                  startLength={6}
+                  className={`tw-text-secondary hover:tw-underline ${
+                    delegateLoading && 'tw-skeleton'
+                  }`}
```
Comment (tiltom) [inline src/app/containers/StakePage/components/VestingContract.tsx:205]:
```suggestion
                  className={classNames('tw-text-secondary hover:tw-underline'), {
                    'tw-skeleton': delegateLoading
                  }) }
```

File: src/app/containers/StakePage/components/VestingContract.tsx [hunk 4]
```diff
@@ -168,145 +168,134 @@ export function VestingContract(props: Props) {
...
+              </>
+            )}
+            {!delegate.length && (
+              <>{t(translations.stake.delegation.noDelegate)}</>
+            )}
+          </p>
+        </td>
+        <td className="tw-text-left tw-hidden lg:tw-table-cell tw-font-normal">
+          {locked && (
+            <p className={`tw-m-0 ${!unlockDate && 'tw-skeleton'}`}>
```
Comment (tiltom) [inline src/app/containers/StakePage/components/VestingContract.tsx:216]:
```suggestion
            <p className={classNames('tw-m-0', {
            'tw-skeleton': !unlockDate}
            )}>
```

File: src/app/containers/StakePage/components/VestingContract.tsx [hunk 5]
```diff
@@ -168,145 +168,134 @@ export function VestingContract(props: Props) {
...
+              {Math.abs(dayjs().diff(parseInt(unlockDate) * 1e3, 'days'))}{' '}
+              {t(translations.stake.days)}
+            </p>
+          )}
+        </td>
+        <td className="tw-text-left tw-hidden lg:tw-table-cell tw-font-normal">
+          <p className={`tw-m-0 ${!stakingPeriodStart && 'tw-skeleton'}`}>
```
Comment (tiltom) [inline src/app/containers/StakePage/components/VestingContract.tsx:223]:
```suggestion
          <p className={classNames('tw-m-0', {
          'tw-skeleton': !stakingPeriodStart}
          )}>
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — DISMISSED** [review]

**Review by ShynRou — DISMISSED** [review]

**Review by soulBit — APPROVED** [review]

**Review by JoeStorm99 — APPROVED** [review]

**Review by soulBit — CHANGES_REQUESTED** [review]
Re-reviewed this and noticed that withdraw buttons are missing for some assets in the Fee Claiming block (tested this on mainnet):

image

**Comment by creed-victor** [comment]
> Re-reviewed this and noticed that withdraw buttons are missing for some assets in the Fee Claiming block (tested this on mainnet)

Amount to withdraw is 0, do you prefer on still being able to see the button?

**Review by JoeStorm99 — DISMISSED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**